src/main/java/seedu/address/logic/commands/DeleteCommentCommand.java

```diff
@@ -0,0 +1,118 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ASSIGNMENT;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.gradedtest.GradedTest;
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.assignment.AssignmentMap;
+import seedu.address.model.person.assignment.AssignmentName;
+import seedu.address.model.person.assignment.Comment;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Deletes the comment from a person's assignment.
+ */
+public class DeleteCommentCommand extends Command {
+
+    public static final String COMMAND_WORD = "deletecomment";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Deletes the comment from a person’s assignment "
+            + "identified by the index number used in the displayed person list. "
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_ASSIGNMENT + "ASSIGNMENT ";
+
+    public static final String MESSAGE_SUCCESS = "Deleted comment from assignment: %1$s";
+
+    public static final String MESSAGE_CONSTRAINT = "Cannot delete the comment from an assignment with no comment";
+
+    private final AssignmentName assignmentName;
+    private final Index index;
+
+    /**
+     * Creates a DeleteCommentCommand to delete the comment from a person's assignment.
+     */
+    public DeleteCommentCommand(Index index, AssignmentName assignmentName) {
+        requireAllNonNull(index, assignmentName);
+        this.index = index;
+        this.assignmentName = assignmentName;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        if (!AssignmentName.isValidName(this.assignmentName.toString())) {
```

Comment:
perhaps you want to change the parameters of `isValidName` to accept a AssignmentName instead? if not it seems like you're circling around by converting it into a String and then back into AssignmentName in the method